app/src/main/kotlin/com/wire/android/ui/calling/OngoingCallScreen.kt

```diff
@@ -75,19 +75,23 @@ private fun OngoingCallTopBar(
 }
 
 @Composable
-private fun CallingControls(ongoingOngoingCallViewModel: OngoingCallViewModel = hiltViewModel()) {
+private fun CallingControls(ongoingCallViewModel: OngoingCallViewModel = hiltViewModel()) {
```

Comment:
like in the case in the previous PR i think it makes sense to move this hiltVIewModel() creation up to CallingScreen, since later on you will need to do it anyway :P and pass here lambdas as event receivers

Reply:
I don't want to make `OngoingCallContent` as a God method, let's keep it here :)

Reply:
and why would it be an issue you think ? you could call the viewmodel methods here : 

```
 sheetContent = {
            CallingControls( 
            muteCall = viewModel::muteCall,
            hangUp = viewModel::hangUp,
            ...
            .)
        },
```

Reply:
also I see that you call this :
'''  ongoingCallViewModel: OngoingCallViewModel = hiltViewModel() ''' 
 twice in the OngoingCallScreen(), this is going to work correctly and return you the same instance of the viewmodel since you are in the scope of the NavBackStackEntry, it add unneccessary overhead and unnecessary code execution in my opinion, since you already have this OngoingCallViewModel  in the scope of OngoingCallScreen() 
 
 I think OngoingCallScreen() should be responsible of passing the events and state to the nested Composable functions and they should lamba's and some dumb primities values, I even think that since you are doing this : 
 
'''
 @Composable
private fun CallingControls(ongoingCallViewModel: OngoingCallViewModel = hiltViewModel()) 
''' 
the CallingControls will be recomposed each time you change the OngoingCallState inside the viewmodel ( i am not sure about that one ) because Compose has not really a way of comparing if it should recompose CallingControls method if you are passing there a instance of hiltViewModel, hence you are not using the benefit of skipping the recomposition, but I am not 100 % of that, because it will return the same instance of ViewModel and I think the hashcode should be the same, but since it is not annotated as @Stable it will not be able to invalidate the change

Reply:
all in all I see more disadvantes than advantages of doing it that way honestly :P unless I am saying some bullshit of course ;P

Reply:
That make a sense 👍